path_service_principal_test.go

```diff
@@ -1019,7 +1022,7 @@ func TestCredentialInteg_msgraph(t *testing.T) {
 		// Verify that SP get is an error after delete. Expected there
 		// to be a delay and that this step would take some time/retries,
 		// but that seems not to be the case.
-		assertServicePrincipalDoesNotExist(t, provider.spClient, spObjID)
+		assertServicePrincipalExistence(t, provider.spClient, spObjID, false)
```

Comment:
Quick suggestion unrelated to changes: we should check the error on `b.spRevoke` above similar to what's done in `TestRoleAssignmentWALRollback`. Will make it more obvious what failed for future test runs. Thanks!

Reply:
Good call, added a error check on spRevoke